### PR TITLE
feat(shared): add block item wrapper

### PR DESCRIPTION
**Pull request by fulopdaniel**
1. Add Block Item Wrapper to shared folder. It wraps an item around with a purple outline and provides a toolbar. The toolbar is customisable via props.

2. Move BlockInjectButton to its own folder.

3. Do & Dont Drag and Drop was very laggy. This was because the RTE parsing is a very heavy operation and was run on every rerender. Wrapped it inside callbacks so it only recalculates if needed.

4. Use this new Block Item Wrapper in Do & Dont.

### CODE REVIEW

**Review by Walkeezy — APPROVED** [review]
Looks good to me and works well 👍🏻

File: packages/dos-donts-block/src/DoDontItem.tsx [hunk 1]
```diff
@@ -143,153 +155,169 @@ export const DoDontItem = React.forwardRef<HTMLDivElement, DoDontItemProps>(
...
+                                : []),
+                            ...(!!linkedImage
+                                ? [
+                                      {
+                                          title: 'Replace with asset',
+                                          icon: <IconImageStack20 />,
+                                          onClick: onOpenAssetChooser,
+                                      },
+                                  ]
+                                : []),
```
Comment (celine-s) [inline packages/dos-donts-block/src/DoDontItem.tsx:206]:
```suggestion
                        [
                            ...(!!linkedImage
                                ? [
                                      {
                                          title: 'Replace with upload',
                                          icon: <IconArrowCircleUp20 />,
                                          onClick: onUploadClick,
                                      },
                                      {
                                          title: 'Replace with asset',
                                          icon: <IconImageStack20 />,
                                          onClick: onOpenAssetChooser,
                                      },
                                  ]
                                : []),
```

Comment (celine-s) [reply packages/dos-donts-block/src/DoDontItem.tsx:206]: wouldn't this be the same?

File: packages/dos-donts-block/src/DoDontItem.tsx [hunk 2]
```diff
@@ -143,153 +155,169 @@ export const DoDontItem = React.forwardRef<HTMLDivElement, DoDontItemProps>(
...
-                                    onTextChange={onBodyTextChange}
-                                    placeholder="Add a description"
+                        <div className="tw-w-full tw-flex tw-items-center">
+                            <h3 style={{ marginBottom: 0, width: '100%' }}>
+                                <textarea
+                                    rows={1}
+                                    ref={titleRef}
+                                    onChange={(event) => {
+                                        setInternalTitle(event.target.value);
+                                    }}
```
Comment (celine-s) [inline packages/dos-donts-block/src/DoDontItem.tsx:277]:
```suggestion
                                    onChange={(event) => setInternalTitle(event.target.value)}
```

**Review by celine-s — APPROVED** [review]
just some nits. Component looks good to me and runs smoothly

**Pull request merged**